cmd/osm-controller/osm-controller.go

```diff
@@ -189,18 +193,27 @@ func main() {
 		events.GenericEventRecorder().FatalEvent(err, events.InitializationError, "Error creating MeshSpec")
 	}
 
-	certManager, certDebugger, err := getCertificateManager(kubeClient, kubeConfig, cfg)
+	// Intitialize certificate manager/provider
+	certProviderUtil := &certproviderutil.CertProviderUtil{
+		KubeClient:            kubeClient,
+		KubeConfig:            kubeConfig,
+		Cfg:                   cfg,
+		CertProviderKind:      osmCertificateManagerKind,
+		CertProviderNamespace: osmNamespace,
+		CABundleSecretName:    caBundleSecretName,
+
+		VaultOptions:       vaultOptions,
+		CertManagerOptions: certManagerOptions,
+	}
+
+	certManager, certDebugger, err := certProviderUtil.GetCertificateManager()
```

Comment:
This almost looks like it could be `NewCertificateProfider(kubeClient, kubeConfig, cfg ......)`, which much like `GetCertificateManager` returns the cerManager (which I propose we rename to `certProvider`) and the certDebugger

Reply:
I implemented it as such, but also provided another API to retrieve the config, since many tests are relying on retrieving `certificate.Manager` via the k8s secret, which can be retrieved just based on the config.